BranchMetrics/android-branch-deep-linking-attribution
# add tune migration value (5) for the update key in open/install requests

## Reference
SDK-927 -- Detect TUNE migrations, send "5" in the payload field "update"

## Description
There is a field "update" in the payload, the SDK reports values 0-2 in that field:

private static final int STATE_FRESH_INSTALL = 0;
private static final int STATE_UPDATE = 2;
private static final int STATE_NO_CHANGE = 1;
The server has its own logic that converts it to OPEN/INSTALL/REINSTALL (Open/Install/Reinstall Logic from API Open )

When an app migrates from Tune to Branch, the Android SDK currently report this as STATE_FRESH_INSTALL. The server then marks this as REINSTALL. What we want is for the server to mark this session as OPEN. iOS implemented this by sending value 5 in the case of Tune migration.

## Testing Instructions
Add Tune SDK to sample app, run it, replace TUNE SDK with Branch SDK, bump up app version, run it. Observe update=5 in the payload.

## Risk Assessment [`HIGH` || `MEDIUM` || `LOW`]
MEDIUM


## Reviewer Checklist (To be checked off by the reviewer only)

- Correctness & Style
